cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -708,239 +714,268 @@ func TestStartDeletion(t *testing.T) {
 				"empty-but-with-pods-node-1": {ResultType: status.NodeDeleteErrorInternal, Err: cmpopts.AnyError},
 			},
 		},
-	} {
-		t.Run(tn, func(t *testing.T) {
-			// This is needed because the tested code starts goroutines that can technically live longer than the execution
-			// of a single test case, and the goroutines eventually access tc in fakeClient hooks below.
-			tc := tc
-			// Insert all nodes into a map to support live node updates and GETs.
-			nodesByName := make(map[string]*apiv1.Node)
-			nodesLock := sync.Mutex{}
-			for _, node := range tc.emptyNodes {
-				nodesByName[node.Name] = node
-			}
-			for _, node := range tc.drainNodes {
-				nodesByName[node.Name] = node
-			}
+	}
 
-			// Set up a fake k8s client to hook and verify certain actions.
-			fakeClient := &fake.Clientset{}
-			type nodeTaints struct {
-				nodeName string
-				taints   []apiv1.Taint
-			}
-			taintUpdates := make(chan nodeTaints, 10)
-			deletedNodes := make(chan string, 10)
-			deletedPods := make(chan string, 10)
+	return testCases
+}
 
-			ds := generateDaemonSet()
+func TestStartDeletion(t *testing.T) {
+	testNg1 := testprovider.NewTestNodeGroup("test-ng-1", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	testNg1.SetOptions(&config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: false,
+	})
+	testNg2 := testprovider.NewTestNodeGroup("test-ng-2", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	testNg2.SetOptions(&config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: true,
+	})
 
-			// We're faking the whole k8s client, and some of the code needs to get live nodes and pods, so GET on nodes and pods has to be set up.
-			fakeClient.Fake.AddReactor("get", "nodes", func(action core.Action) (bool, runtime.Object, error) {
-				nodesLock.Lock()
-				defer nodesLock.Unlock()
-				getAction := action.(core.GetAction)
-				node, found := nodesByName[getAction.GetName()]
-				if !found {
-					return true, nil, fmt.Errorf("node %q not found", getAction.GetName())
+	testSets := []testSet{
+		{
+			nodeGroup: testNg1,
+			testCases: getStartDeletionTestCases(testNg1),
+		},
+		{
+			nodeGroup: testNg2,
+			testCases: getStartDeletionTestCases(testNg2),
```

Comment:
I am running the test cases twice one for each nodegroup defined above where in the first nodegroup, `IgnoreDaemonSetsUtilization: false` and for the second nodegroup `IgnoreDaemonSetsUtilization: true`. This is to test `IgnoreDaemonSetsUtilization` option on the nodegroup which is called [here](https://github.com/kubernetes/autoscaler/pull/5672/files#diff-73af8b1e646ba0094f2cc947fc77c027e52714908ed9b6452c6b9cd577229beeR320).